idseq_dag/steps/blast_contigs.py

```diff
@@ -433,6 +439,47 @@ def get_top_m8_nr(orig_m8, blast_top_m8):
             if top_line is not None:
                 top_m8f.write(top_line)  # Unify reranked formats for NT and NR
 
+
+    @staticmethod
```

Comment:
This filter-and-group-by logic is exactly the same as before, just pulled out of the main function so that instead of accumulating the groups in a dictionary, it would yield them one at a time.